clang/docs/ClangSYCLLinkWrapper.rst

```diff
@@ -0,0 +1,67 @@
+====================
+Clang SYCL link Wrapper
+====================
+
+.. contents::
+   :local:
+
+.. _clang-sycl-link-wrapper:
+
+Introduction
+============
+
+This tool works as a wrapper around the SYCL device code linking process.
+Purpose of this wrapper is to provide an interface to link SYCL device bitcode
+in LLVM IR format, run some SYCL-specific finalization steps and then use the
+SPIR-V LLVM Translator tool to produce the final output.
+
+Device code linking for SYCL offloading kind has a number of known quirks that
+makes it difficult to use in a unified offloading setting. Two of the primary
+issues are:
+1. Several finalization steps are required to be run on the fully-linked LLVM
+IR bitcode to gaurantee conformance to SYCL standards. This step is unique to
+SYCL offloading compilation flow.
+2. SPIR-V LLVM Translator tool is an extenal tool and hence SPIR-V IR code
+generation cannot be done as part of LTO. This limitation will be lifted once
+SPIR-V backend is available as a viable LLVM backend.
+
+This tool works around these issues.
+
+Usage
+=====
+
+This tool can be used with the following options. Several of these options will
+be passed down to downstrea tools like 'llvm-link', 'llvm-spirv', etc.
+
+.. code-block:: console
+
+  OVERVIEW: A utility that wraps around the SYCl device code linking process.
+  This enables linking and code generation for SPIR-V JIT targets.
```

Comment:
What about AOT targets?

Reply:
I was planning to add it as the implementation evolves in near future. But I will add it now and provide a PR comment about upcoming changes.

Thanks